### DISCUSSION TITLE
Taskbar button width is too large

### DISCUSSION THREAD

**Discussion by lmqferreira** [valinet/ExplorerPatcher#664] [General]
The taskbar buttons in a 4k display (150% scale) are too large. I suspect, but I'm not absolutely positive that this behavior started after the installation of the latest update to Windows 11 build 10.0.22000.434.
<img width="1664" alt="taskbar" src="https://user-images.githubusercontent.com/14849337/149389641-d3dc3194-5226-424c-b539-8ba4bbc03121.png">

**Comment by lmqferreira** [comment]
<img width="1619" alt="taskbar2" src="https://user-images.githubusercontent.com/14849337/149491083-18f02431-932f-4744-af0c-afe213cba653.png">
It seems that there is no limit to the width of the buttons and they are filling the available space. Is this something dependent on ExplorerPatch or a bug in explorer.exe itself? Thanks! Congrats for this great work!

**Reply by valinet** [reply]
It does not happen on my end, at least not all the times. It sometimes happens when undocking the laptop, I think, when the DPI changes or something like that. Restart the computer and it should be fixed. Idk, it's a bug in `explorer`, idk where it comes from. Maybe limiting the button size through the registry helps (REG_SZ `MinWidth` in `HKEY_CURRENT_USER\Control Panel\Desktop\WindowMetrics`, value 38-500)...?

**Reply by lmqferreira** [reply]
That's exactly that what happened. I was just to post that found the solution at https://www.yuenx.com/2021/taskbar-icon-button-width-too-wide-in-windows-10/. For some reason that Registry key is not there. Perhaps ExplorerPatch could have a feature to define that... just a suggestion. Again, congrats for the great work. This utility is really helpful for power users :)

**Reply by valinet** [reply]
The setting's not really needed at all times. Idk, yeah, this has to be investigated more thoroughly, in the future, to identify exactly what triggers it and what a proper solution is. For now, yeah, it seems this workaround works. It's the first result on Google, that's where I got the info from too.

> Again, congrats for the great work.

Thanks.

**Reply by valinet** [reply]
Yeah, just adding that registry entry (`MinWidth`=38) in case nothing's in there seems to solve this issue. I'll try with that, have EP create that at startup if nothing's already in there, I don't think it can otherwise hurt.

**Reply by valinet** [reply]
Yeah, I pushed a "fix" that creates that registry entry with value 38 if not already in there with some other value. Hopefully, it doesn't break stuff for other users, we'll see the reports.

**Comment by gfernandez-me** [comment]
I had the same issue, also an option to hide texts from the task buttons would be awesome
but after logout/restart will solve the issue as far @valinet  added the 38 width automatically in the app

**Reply by Poopooracoocoo** [reply]
>also an option to hide texts from the task buttons would be awesome

If you don't want the items to combine but hide labels, follow this: https://github.com/valinet/ExplorerPatcher/issues/767#issuecomment-1027926850
Hope this helps!

**Reply by valinet** [reply]
You are talking about 3 different things in one sentence:

* MinWidth 38 is so that it mitigates a bug where buttons on the taskbar become too large, but it breaks minimized buttons in MDI apps
* The issue above was something else (device stuck in tablet mode) and I solved that and provided steps to get it right with the latest version
* To hide taskbar items labels, it’s right there in front of you eyes in Properties - Taskbar - Combine items settings. I don’t get how so many people miss that. Also, what’s the point of using the old taskbar without labels, since the new taskbar is exactly that basically.

**Reply by OrthodoxWindows** [reply]
I confirm, MinWidth 38 break the minimized windows in MDI applications, what is worrying when Mdi applications regularly use.
Use the old taskbar without the labels has many advantages; Among these, the possibility of benefiting Open-Shell and 7+ TaskBar Tweaker's personalization options (especialy the possibility of not grouping icons).

**Comment by Pirateguybrush** [comment]
Hey there,

Setting the minwidth registry key is messing with my secondary monitor's taskbar, for which I'm using displayfusion.  Can this please be made into a toggleable option so I don't have to keep deleting the key?

**Reply by valinet** [reply]
I will make it optional, yes.
